swap_meet/vendor.py

```diff
@@ -0,0 +1,72 @@
+from .item import Item
+
+class Vendor:
+    def __init__(self, inventory = None):
+        if inventory is None:
+            self.inventory = []
+        else:
+            self.inventory = inventory
+
+    def add(self, item):
+        self.inventory.append(item)
+        return item
+
+    def remove(self, item):
+        if item in self.inventory:
+            self.inventory.remove(item)
+            return item
+        return False
+
+    def get_by_category(self, category):
+        items = [item for item in self.inventory if item.category == category]
+        return items
+
+    def swap_items(self, other, my_item, their_item):
+        if my_item in self.inventory and their_item in other.inventory:
+            other.add(self.remove(my_item))
+            self.add(other.remove(their_item))
```

Comment:
Yay method chaining! As long as you understand what this is doing this is a great, more advanced way to write this code more succinctly!